file_util/tests/directory_utils_test.cc

```diff
@@ -0,0 +1,82 @@
+#include <catch.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <leatherman/file_util/directory.hpp>
+
+#ifdef _WIN32
+#include <windows.h>
```

Comment:
Let's pair up on the test cases, I think there's a number of things that can be improved.